### PR TITLE
[WIP] Drop Scala 2.11

### PR DESCRIPTION
**Pull request by sideeffffect**
It would be great to make one more release before we merge this. One last time for Scala 2.11 :smiling_face_with_tear: 

Scala 2.11 has been dropped by both

Scala.js

 * https://github.com/scala-js/scala-js/issues/4759
 * https://contributors.scala-lang.org/t/we-are-considering-dropping-support-for-scala-2-11-in-scala-js-core/6016
 * https://www.scala-js.org/news/2023/01/26/announcing-scalajs-1.13.0/

Scala Native

 * https://github.com/scala-native/scala-native/issues/2780
 * https://scala-native.org/en/stable/changelog/0.4.10.html

### CODE REVIEW

File: core-tests/shared/src/test/scala/zio/ExecutorSpec.scala [hunk 1]
```diff
@@ -35,7 +35,7 @@ object TestExecutor {
     override def reportFailure(cause: Throwable): Unit = ()
   }
 
-  // backward compatibility for scala 2.11.12
+  // backward compatibility for scala 2.11.12 // TODO: remove?
```
Comment (sideeffffect) [inline core-tests/shared/src/test/scala/zio/ExecutorSpec.scala:38]: What to do with this?

File: test/shared/src/main/scala/zio/test/TestOutput.scala [hunk 2]
```diff
@@ -139,7 +139,7 @@ object TestOutput {
         )
       }
 
-    // We need this helper to run on Scala 2.11
+    // We need this helper to run on Scala 2.11 // TODO: remove?
```
Comment (sideeffffect) [inline test/shared/src/main/scala/zio/test/TestOutput.scala:142]: What to do with this?

**Comment by sideeffffect** [comment]
@adamgfraser @mijicd can we please try rerunning the failed job? :pray:

**Comment by adamgfraser** [comment]
Why are we doing this?

**Comment by sideeffffect** [comment]
Oh, of course, I should have explained it in the beginning. I updated the PR description.

**Comment by sideeffffect** [comment]
`    - promise ugly path test` still fails :crying_cat_face: any idea why that would be the case?

**Comment by adamgfraser** [comment]
Would like to merge this with support for Scala Native. Otherwise this is pure downside.

**Comment by adamgfraser** [comment]
Will take a look.

**Comment by jdegoes** [comment]
Fixed by @adamgfraser's PR. Thanks for your work on this!

**Pull request closed without merging**